src/gui/trackerlogin.cpp

```diff
@@ -39,24 +40,33 @@ trackerLogin::trackerLogin(QWidget *parent, BitTorrent::TorrentHandle *const tor
 {
   setupUi(this);
   setAttribute(Qt::WA_DeleteOnClose);
+
+  buttonBox->button(QDialogButtonBox::Ok)->setText(tr("Log in"));
+
   login_logo->setPixmap(QPixmap(QString::fromUtf8(":/icons/qbt-theme/encrypted.png")));
+
   tracker_url->setText(torrent->currentTracker());
+
+  connect(buttonBox->button(QDialogButtonBox::Ok), &QPushButton::clicked, this, &trackerLogin::loginButtonClicked);
+  connect(buttonBox->button(QDialogButtonBox::Cancel), &QPushButton::clicked, this, &trackerLogin::cancelButtonClicked);
+  connect(linePasswd, &QLineEdit::returnPressed, this, &trackerLogin::loginButtonClicked);
   connect(this, SIGNAL(trackerLoginCancelled(QPair<BitTorrent::TorrentHandle*, QString>)), parent, SLOT(addUnauthenticatedTracker(QPair<BitTorrent::TorrentHandle*, QString>)));
```

Comment:
Maybe you want to change this connect style too?

Reply:
a little tricky, I'll skip it.

Reply:
Some smart-ass connection... Need to check it out. Apparently, this is the heritage of the past. Can you put TODO comment here?

Reply:
ok.

Reply:
"Tracker login" feature looks deprecated...
I think we should completely remove all appropriate code and files from qBittorrent in v4.0.
@Chocobo1, @evsh, what do you think about?

Reply:
Is it replaced by cookie manager?

Reply:
> "Tracker login" feature looks deprecated...

Sorry, I did not follow this subject. Let's remove it when so.

Reply:
It was used by libtorrent to pass login info to the trackers. From libtorrent code:
>// sets a username and password that will be sent along in the HTTP-request
// of the tracker announce. Set this if the tracker requires authorization.

The username and password were sent as-is, without any encryption.
It's deprecated in libtorrent-1.1 so I think we should remove it as well.

Reply:
Anyway, it's separate issue.